sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -958,6 +958,20 @@ case class Sort(
   override protected def withNewChildInternal(newChild: LogicalPlan): Sort = copy(child = newChild)
 }
 
+/**
+ * A special Sort node whose underlying Sort would not be eliminated
+ * by [[org.apache.spark.sql.catalyst.optimizer.EliminateSorts]].
+ */
+case class NoEliminateSort(
```

Comment:
I'll probably call it `Clustering`, as it's the intention of the v1 write rule: we want the data to be clustered in a certain way.

Reply:
Addressed.